README.md

```diff
@@ -1,154 +1,62 @@
+* [Edgelesssys Constellation aTLS](https://github.com/edgelesssys/constellation/tree/main/internal/atls)
+* [SGX Attested TLS](https://github.com/intel/linux-sgx/tree/master/SampleCode/SampleAttestedTLS)
+* [OpenEnclave Attested TLS](https://github.com/openenclave/openenclave/blob/master/samples/attested_tls/AttestedTLSREADME.md)
+* [Gramine RA-TLS](https://github.com/gramineproject/gramine/tree/master/tools/sgx/ra-tls)
+* [Inclave Containers RATS-TLS](https://github.com/inclavare-containers/rats-tls/blob/master/docs/design/design.md)
+
+There is also an effort to standardize and unify this approach over the existing Confidential Computing (CC) technologies:
+* [Using Attestation in Transport Layer Security (TLS) and Datagram Transport Layer Security (DTLS) IETF draft](https://datatracker.ietf.org/doc/draft-fossati-tls-attestation/), the reference implementation based on Mbed TLS can be found here [tls-attestation](https://github.com/hannestschofenig/mbedtls/tree/tls-attestation)
+* [Confidential Computing Consortium Interoperable TLS](https://github.com/CCC-Attestation/interoperable-ra-tls), [Unifying Remote Attestation Protocol Implementations (blogpost)](https://confidentialcomputing.io/2023/03/06/unifying-remote-attestation-protocol-implementations/)
+
+## RA-TLS details
+
+The RA-TLS implementation is based on the ideas used in SGX Attested TLS and Edgelesssys Constellation aTLS projects. The RA-TLS library contained in this repository utilizes the TLS layer from the RusTLS library and performs remote attestation by providing custom certificate resolvers and verifiers. During the startup, a Realm application generates an asymmetric key pair (a private key and a corresponding public key). The keys will be used to generate a self-signed certificate. During the TLS handshake, a Verifier (or a Reliant Party) sends a request containing the randomly generated nonce. The Realm application calculates a challenge as a hash of the nonce (received from the requester) and the public key (challenge = Hash(nonce || Public Key)). Then, the application requests the CCA attestation token by providing challenge. The resulting attestation token is embedded into the self-signed certificate. Next, the certificate is sent to the requester. The requester verifies the self-signed certificate, checks the cryptographic binding between the attestation token and the certificate and verifies the content of the token. The negative appraisal of the CCA attestation token causes a disconnection.
+
+![](./doc/ratls-cert.png)
+X.509 certificate with the embedded CCA attestation token
+
+### Use cases
+The library can be utilized in three use cases (notice, we are using here the same terminology as in [RFC 8446](https://www.rfc-editor.org/rfc/rfc8446.html#section-1.1) i.e. the client just means "The endpoint initiating the TLS connection.", the server is defined as: "The endpoint that did not initiate the TLS connection"):
+![](./doc/use1.png)
```

Comment:
Those images are in the wrong places I think.

use3 definitely looks like mutual attestation.

use1 and use2 should be in their appropriate sections as well.